### PR TITLE
softpoweroff: 1110 rebase for softoff application

### PR DESCRIPTION
**Pull request by sagisin**
Rebase for the soft off commits into 1110.

### CODE REVIEW

File: softoff/softoff.cpp [hunk 1]
```diff
@@ -37,10 +37,10 @@ pldm::pdr::TerminusID TID = 0;
 
 namespace sdbusRule = sdbusplus::bus::match::rules;
 
-SoftPowerOff::SoftPowerOff(sdbusplus::bus_t& bus, sd_event* event,
-                           pldm::InstanceIdDb& instanceIdDb) :
+SoftPowerOff::SoftPowerOff(sdbusplus::bus::bus& bus, sd_event* event,
```
Comment (manojkiraneda) [inline softoff/softoff.cpp:40]: this is a bad merge, need this to be reverted back to `bus_t` as per master.

File: softoff/softoff.cpp [hunk 2]
```diff
@@ -137,12 +137,17 @@ void SoftPowerOff::hostSoftOffComplete(sdbusplus::message_t& msg)
     if (msgSensorID == sensorID && msgSensorOffset == sensorOffset &&
         msgEventState == PLDM_SW_TERM_GRACEFUL_SHUTDOWN && msgTID == TID)
     {
-        // Receive Graceful shutdown completion event message. Disable the timer
-        auto rc = timer.stop();
-        if (rc < 0)
+        std::cout << "Recieved the graceful shutdown signal from host\n";
```
Comment (manojkiraneda) [inline softoff/softoff.cpp:140]: we already moved to lg2 right ?

File: softoff/softoff.cpp [hunk 3]
```diff
@@ -362,26 +367,37 @@ int SoftPowerOff::hostSoftOff(sdeventplus::Event& event)
...
-        else
+        if (!noTimeOut)
         {
-            error(
-                "Timer started waiting for host soft off, TIMEOUT_IN_SEC = {TIMEOUT_SEC}",
-                "TIMEOUT_SEC", SOFTOFF_TIMEOUT_SECONDS);
+            // Start Timer
+            using namespace std::chrono;
+            auto timeMicroseconds =
+                duration_cast<microseconds>(seconds(SOFTOFF_TIMEOUT_SECONDS));
```
Comment (manojkiraneda) [inline softoff/softoff.cpp:385]: is this really needed to move under the `if` ?

File: .secrets.baseline [hunk 4]
```diff
@@ -0,0 +1,130 @@
+{
```
Comment (manojkiraneda) [inline .secrets.baseline:1]: is this file needed ? for every folder ? i thought its just needed for the root folder.

File: softoff/main.cpp [hunk 5]
```diff
@@ -49,6 +67,23 @@ int main()
...
+        method.append(
+            std::vector<
+                std::pair<std::string, std::variant<std::string, uint64_t>>>());
+        try
+        {
+            bus.call_noreply(method);
+        }
+        catch (const sdbusplus::exception::exception& e)
+        {
+            std::cerr << "SoftPowerOff:Failed to create BMC dump, ERROR="
```
Comment (manojkiraneda) [inline softoff/main.cpp:83]: lg2 missing

File: softoff/meson.build [hunk 6]
```diff
@@ -16,12 +16,16 @@ executable('pldm-softpoweroff',source,
...
 
-if get_option('systemd').allowed()
-  filesystem.copyfile(
-    'services/pldmSoftPowerOff.service',
-    'pldmSoftPowerOff.service',
+if get_option('systemd').enabled()
+  configure_file(input: 'services/pldmSoftPowerOff.service',
+                   output: 'pldmSoftPowerOff.service',
+                   copy: true,
+                   install_dir: systemd_system_unit_dir)
```
Comment (manojkiraneda) [inline softoff/meson.build:23]: we need just master change here, no need to put the downstream change.

File: softoff/softoff.cpp [hunk 7]
```diff
@@ -362,26 +367,37 @@ int SoftPowerOff::hostSoftOff(sdeventplus::Event& event)
             exit(-1);
         }
 
-        responseReceived = true;
+        error(
+            "Got the response for set effecter states command, PLDM RC = {RC}",
+            "RC", lg2::hex, static_cast<uint16_t>(response->payload[0]));
```
Comment (manojkiraneda) [inline softoff/softoff.cpp:372]: why is this an error ?

File: softoff/softoff.cpp [hunk 8]
```diff
@@ -397,6 +413,8 @@ int SoftPowerOff::hostSoftOff(sdeventplus::Event& event)
             "RC", static_cast<int>(rc), "ERR", errno);
         return PLDM_ERROR;
     }
+    std::vector<uint8_t> requestbuffer(requestMsg.begin(), requestMsg.end());
+    pldm::utils::printBuffer(pldm::utils::Tx, requestbuffer);
```
Comment (manojkiraneda) [inline softoff/softoff.cpp:417]: is this change not upstreamed ? should not we upstream ?

File: softoff/softoff.hpp [hunk 9]
```diff
@@ -26,9 +26,11 @@ class SoftPowerOff
      *  @param[in] bus       - system D-Bus handler
      *  @param[in] event     - sd_event handler
      *  @param[in] instanceDb - pldm instance database
+     *  @param[in] noTimeOut - bool variable that captures
+     *                         is there is indeed a timeout
      */
-    SoftPowerOff(sdbusplus::bus_t& bus, sd_event* event,
-                 InstanceIdDb& instanceIdDb);
+    SoftPowerOff(sdbusplus::bus::bus& bus, sd_event* event,
```
Comment (manojkiraneda) [inline softoff/softoff.hpp:32]: need it to be changed back to bus_t

**Review by manojkiraneda — CHANGES_REQUESTED** [review]

File: softoff/meson.build (hunk 6 above)
Comment (sagisin) [reply softoff/meson.build:23]: done

File: softoff/main.cpp (hunk 5 above)
Comment (sagisin) [reply softoff/main.cpp:83]: done

File: softoff/softoff.cpp (hunk 2 above)
Comment (sagisin) [reply softoff/softoff.cpp]: done

File: softoff/softoff.cpp (hunk 7 above)
Comment (sagisin) [reply softoff/softoff.cpp]: done

File: softoff/softoff.cpp (hunk 3 above)
Comment (sagisin) [reply softoff/softoff.cpp:385]: yes, we do this only if (!noTimeOut)

File: softoff/softoff.cpp (hunk 8 above)
Comment (sagisin) [reply softoff/softoff.cpp:417]: nope, will create a gerrit patch for this.

File: softoff/softoff.hpp (hunk 9 above)
Comment (sagisin) [reply softoff/softoff.hpp]: done
